pyomo/contrib/doe/measurements.py

```diff
@@ -25,252 +25,270 @@
+            all_index_for_var = list(itertools.product(*all_index_list))
+
+            for lst in all_index_for_var:
+                name1 = name_data+"["
+                for i, idx in enumerate(lst):
+                    name1 += str(idx)
+
+                    # if i is the last index, close the []. if not, add a "," for the next index. 
+                    if i==len(lst)-1:
+                        name1 += "]"
+                    else:
+                        name1 += ","
+
+                self.special_set.append(name1)
+
+
+    def _names(self, var_name, extra_index, time_index, values, upper_bound, lower_bound, check_values_bounds=False):
+        """
+        Check if the measurement information provided are valid to use. 
+        """
+        num_var = len(var_name)
+        
+        if extra_index and len(extra_index) != num_var: 
+            raise ValueError("Extra_index is of different length with var_name. ")
+
+        if len(time_index) != num_var:
+            raise ValueError("time_index is of different length with var_name.")
+
+        if check_values_bounds:
+            num_name = len(self.special_set)
+
+            if values and len(values) != num_name:
+                raise ValueError("Values is of different length with var_name.")
+            
+            if upper_bound and type(upper_bound)==list and len(upper_bound)!= num_name:
+                raise ValueError("Upperbounds is of different length with var_name.")
+            
+            if lower_bound and type(lower_bound)==list and len(lower_bound)!= num_name:
+                raise ValueError("Lowerbounds is of different length with var_name.")
+
+    
+class Measurements(SpecialSet):
+    def __init__(self):
         """
-        flatten_variance = {}
-        for i in flatten_measure_name:
-            if variance is None:
-                flatten_variance[i] = 1
-            else:
-                # split the flattened name if needed
-                if self.ind_string in i:
-                    measure_name = i.split(self.ind_string)[0]
-                    measure_index = i.split(self.ind_string)[1]
-                    if type(name_and_index[measure_name][0]) is int:
-                        measure_index = int(measure_index)
-                    flatten_variance[i] = variance[measure_name][measure_index]
-                else:
-                    flatten_variance[i] = variance[i]
-        self.flatten_variance = flatten_variance
-
-    def _generate_flatten_timeset(self, all_info, flatten_measure_name,name_and_index):
+        This class stores information on which algebraic and differential variables in the Pyomo model are considered measurements. 
         """
-        Generate flatten variables timeset. Return a dict where keys are the flattened variable names,
-        values are a list of measurement time.
+        super().__init__()
 
+    def specify(self, self_define_res):
```

Comment:
How will this function change if the user is going to call "add_variable" once for each different variable?